pandas/tests/test_algos.py

```diff
@@ -326,6 +326,86 @@ def test_factorize_na_sentinel(self, sort, na_sentinel, data, uniques):
         else:
             tm.assert_extension_array_equal(uniques, expected_uniques)
 
+    @pytest.mark.xfail(
+        compat.is_platform_windows(),
+        reason="Windows will be coelced to int32 other than int64",
```

Comment:
huh? what is changed to int32? simply construct the expected with a fixed dtype, we don't want to xfail things.

Reply:
emm, i got pretty weird errors before only on windows builds, and complained the `dtype` is different, while getting `int32` but expected is `int64`, so i assume something is different in windows that makes this different